wormhole-foundation/wormhole-connect
# Move to Vite

This branch moves Connect off `react-app-rewire` and onto [Vite](https://vitejs.dev/).

Advantages of Vite:
- Very fast dev tooling
- Native ESM support
- Great TypeScript support

This is a dependency for #1343 

I've decided to leave `builder` alone for now and let it stay on `create-react-app`. Took a stab at it and saw some weird stylesheet issues.

## How to test it

Check out this branch `experiment-vite`, then

```
npm i
npm run start -w wormhole-connect
```

Vite should start a local dev server with the Connect demo app.

## Dev experience improvements

Cold boot is about 10x faster faster (`development`: ~40 sec, `experiment-vite`: 3-4 sec)

Most of those gains are from using `swc` and `rollup` (Vite's bundler). Curious to see if others have similarly good results!
